### PR TITLE
GFC-541: remove commas when computing size of number

### PR DESCRIPTION
**Pull request by co-category**
…arts of numbers

### CODE REVIEW

File: app/uk/gov/hmrc/gform/validation/ValidationService.scala [hunk 1]
```diff
@@ -442,22 +444,23 @@ class ComponentsValidator(
...
         getError(fieldValue, s"must be at most $maxWhole digits")
       case (WholeShape("-", _, _), _, true) => getError(fieldValue, "must be a positive number")
       case (WholeShape(_, _, _), _, _)      => ().valid
-      case (FractionalShape(_, whole, _, fractional), 0, _) if whole.size > maxWhole && fractional.size > 0 =>
+      case (FractionalShape(_, whole, _, fractional), 0, _)
+          if filterCommas(whole).size > maxWhole && filterCommas(fractional).size > 0 =>
         getError(fieldValue, s"number must be at most $maxWhole whole digits and no decimal fraction")
       case (FractionalShape(_, whole, _, fractional), _, _)
-          if whole.size > maxWhole && fractional.size > maxFractional =>
+          if filterCommas(whole).size > maxWhole && filterCommas(fractional).size > maxFractional =>
```
Comment (lutnos) [inline app/uk/gov/hmrc/gform/validation/ValidationService.scala]: Surely you don't need to filterCommas from the fractional part? No commas are allowed there by FractionalShape

File: app/uk/gov/hmrc/gform/validation/ValidationService.scala [hunk 2]
```diff
@@ -442,22 +444,23 @@ class ComponentsValidator(
...
         getError(
           fieldValue,
           s"number must be at most $maxWhole whole digits and decimal fraction must be at most $maxFractional digits")
-      case (FractionalShape(_, whole, _, _), _, _) if whole.size > maxWhole =>
+      case (FractionalShape(_, whole, _, _), _, _) if filterCommas(whole).size > maxWhole =>
         getError(fieldValue, s"number must be at most $maxWhole whole digits")
-      case (FractionalShape(_, _, _, fractional), 0, _) if fractional.size > 0 =>
+      case (FractionalShape(_, _, _, fractional), 0, _) if filterCommas(fractional).size > 0 =>
```
Comment (lutnos) [inline app/uk/gov/hmrc/gform/validation/ValidationService.scala]: Ditto

File: app/uk/gov/hmrc/gform/validation/ValidationService.scala [hunk 3]
```diff
@@ -442,22 +444,23 @@ class ComponentsValidator(
...
         getError(fieldValue, "must be a whole number")
-      case (FractionalShape(_, _, _, fractional), _, _) if fractional.size > maxFractional =>
+      case (FractionalShape(_, _, _, fractional), _, _) if filterCommas(fractional).size > maxFractional =>
```
Comment (lutnos) [inline app/uk/gov/hmrc/gform/validation/ValidationService.scala]: Ditto

File: test/uk/gov/hmrc/gform/services/NumberValidationSpec.scala [hunk 4]
```diff
@@ -73,6 +73,16 @@ class NumberValidationSpec extends Spec {
     validate(`fieldValue - number`, rawDataFromBrowser).futureValue shouldBe ().valid
   }
 
+  "Number format" should "not consider commas when computing the size of the whole number" in new Test {
```
Comment (josef-vlach) [inline test/uk/gov/hmrc/gform/services/NumberValidationSpec.scala]: I already asked in previous PR, but why we have two test to validate
`"+789,343.34"` and two test to validate `"-789,343.34"`?

Comment (josef-vlach) [reply test/uk/gov/hmrc/gform/services/NumberValidationSpec.scala]: I do not like this way of test. I think [Table-driven property checks](http://www.scalatest.org/user_guide/table_driven_property_checks) would be much better approach for this particular usecase.

**Pull request merged**